tests/lib/rules/no-array-prototype-extensions.js

```diff
@@ -321,7 +321,13 @@ ruleTester.run('no-array-prototype-extensions', rule, {
     },
     {
       code: 'something.compact()',
-      output: null,
+      output: 'something.filter(item => item !== undefined && item !== null)',
+      errors: [{ messageId: 'main', type: 'CallExpression' }],
+    },
+    {
+      // Invalid as compact function doesn't support any params. Autofixer removes all the arguments and include the right callback
+      code: 'something.compact(1, getVal(), 3)',
```

Comment:
If there are unexpected params, we probably want to avoid autofixing, as it indicates there's a mistake (perhaps wrong macro was used) or the code is incomplete (copy-paste that hasn't been updated yet).

Reply:
@tgvrssanthosh let's make sure we have a test case that we don't autofix with unexpected params.

Reply:
I just added a test case.